docs/proposals/ai-studio.md

```diff
@@ -0,0 +1,97 @@
+# Motivation
+
+Today, there is no notion of ordering between the containers. But we know that we have a dependency between
+the client application and the container that is running the model.
+
+The second issue is that there is no concept of starting point for a container so today we rely only on the
+container being started by the container engine and we know that this is not adequate for the model service container
+
+So this is handle by a kind of dirty fix: the containers are all started in parallel but as the client application
+will fail because the model service is started (as it take a while), so we are trying to restart the client application
+until the model service is properly started.
+
+The purpose of this change is to propose an update to the ai-studio.yaml so that it is as much generic as it
+could be and inspired from the Compose specification.
+
+## Proposed changes
+
+Define a dependency between containers: so in the definition of container, we would add a new ```depends``` field
+that would define the ordering between the containers. This would be an optional field.
```

Comment:
I do not think this is necessary, as Kubernetes works without it ?  If a container depends on a database, the container should handle a retry scenario waiting for the database to be available.

I think we should make this file closer to Kubernetes resource definition rather than inventing new concept

Reply:
I don't mind adding the `dependency` flag here if it helps. But we could also enforce some of these checks in the application code itself (confirm there is an available model service before starting the client application). Sounds like that might be better solution?

Reply:
OK so the client application will wait until the model service is available and then starts its logic. So I keep only the healthcheck/readiness one

Reply:
As discussed on slack, the healthcheck will also be helpful on the client container, to AI Studio knows when the complete app is ready for the user

Reply:
Ok, so I can 1)  update the application code for the recipes to wait for a model service to become available. 2) Add a Healthcheck to the service and application Containerfiles? 

is that right? @feloy @jeffmaury ?

Reply:
2) is to be delayed as this changes are not implemented yet. But 1) should be done now without disruption

Reply:
Here is the PR with the proposed change. If ok we can replicate to the other apps.  https://github.com/redhat-et/locallm/pull/41